tools/hilbert.py

```diff
@@ -0,0 +1,589 @@
+#!/usr/bin/env python2
```

Comment:
The `.py` exposes the implementation detail that the script is written in python (at least it suggests that). The user interface should not expose such things. Please add the link.